pkgs/applications/misc/termite/default.nix

```diff
@@ -7,16 +7,17 @@ let
   termite = stdenv.mkDerivation {
     name = "termite-${version}";
 
-    src = fetchgit {
-      url = "https://github.com/thestinger/termite";
-      rev = "refs/tags/v${version}";
+    src = fetchFromGitHub {
+      owner = "thestinger";
+      repo = "termite";
+      rev = "v${version}";
       sha256 = "02cn70ygl93ghhkhs3xdxn5b1yadc255v3yp8cmhhyzsv5027hvj";
+      fetchSubmodules = true;
     };
 
     # https://github.com/thestinger/termite/pull/516
-    patches = [ ./url_regexp_trailing.patch ];
-
-    postPatch = "sed '1i#include <math.h>' -i termite.cc";
+    patches = [ ./url_regexp_trailing.patch
+                ] ++ lib.optional stdenv.isDarwin ./fix_macos.patch;
```

Comment:
This makes the addition of the `<errno.h>` header conditional on using Darwin, please preserve the existing behavior :).

(I suppose either split up the patch or keep the sed line from before and use it unconditionally like we do now?)

Reply:
Whoops, in my tree I've had a copy of the `math.h` sed that adds `errno.h`-- I guess that didn't make it upstream?

Well my apologies about that, but would you be so kind as to add it unconditionally anyway? :)

Reply:
Done in my new commit :)